modules/lang-painless/src/main/java/org/elasticsearch/painless/antlr/Walker.java

```diff
@@ -172,10 +173,12 @@
  */
 public final class Walker extends PainlessParserBaseVisitor<ANode> {
 
-    public static SSource buildPainlessTree(String sourceName, String sourceText, CompilerSettings settings, Printer debugStream) {
-        return new Walker(sourceName, sourceText, settings, debugStream).source;
+    public static SSource buildPainlessTree(MainMethod mainMethod, String sourceName, String sourceText, CompilerSettings settings,
+            Printer debugStream) {
+        return new Walker(mainMethod, sourceName, sourceText, settings, debugStream).source;
```

Comment:
The walker needs to know stuff about the main method so it can figure out the variables in scope.